app/models/terminal_executor.rb

```diff
@@ -31,7 +31,7 @@ def execute!(*commands)
     output, input, pid = PTY.spawn(whitelisted_env, script(commands), options)
     record_pid(pid) do
       begin
-        output.each(256) { |line| @output.write line }
+        output.each(256) { |line| @output.write "[#{DateTime.now.utc.strftime("%T")}] #{line}" }
```

Comment:
Hmm, does this actually work?  'cause the way I read this, that block is chunking output every 256 characters, not every newline.  So this would inject a timestamp every 256 characters, wherever that would be.

`chunk` would be a better name for this variable than `line`.  Unless I'm missing something?